packages/server/graphql/mutations/downgradeToStarter.ts

```diff
@@ -42,6 +48,29 @@ export default {
     }
 
     // RESOLUTION
+    if (!isSuperUser) {
+      const [organizationUsers, organization] = await Promise.all([
+        dataLoader.get('organizationUsersByOrgId').load(orgId),
+        dataLoader.get('organizations').load(orgId)
+      ])
+      const billingLeaderUserId = organizationUsers
+        .filter((organizationUser) => organizationUser.role === 'BILLING_LEADER')
+        .find(({userId}) => userId)!.id
+      const billingLeaderUser = await dataLoader.get('users').load(billingLeaderUserId)
+      if (!billingLeaderUser) {
+        const error = new Error('Unable to find billing leader')
+        sendToSentry(error, {userId: viewerId})
+        return
+      }
+      analytics.organizationDowngraded(viewerId, {
+        orgId,
+        oldTier: tier,
+        newTier: 'starter',
+        billingLeaderEmail: billingLeaderUser.email,
+        orgName: organization.name,
+        reasonsForLeaving
```

Comment:
I added `reasonsForLeaving` as this seems the best place to track this field. I don't love passing this as an arg to the mutation though. 

Alternate options could be sending the segment client mutation from the client rather than recording it here. I could also create an enum of downgrade reasons if reviewers feel the string array is too brittle

Reply:
+1 I would prefer to use enum, otherwise someone can write arbitrary strings?
No strong preference between backend/client.

Reply:
Sure thing, I've added an enum instead